### PR TITLE
[THRILL-6380] - Setup the template for coding challenge

### PR DESCRIPTION
**Pull request by aecordova**
This PR contains the following:
- A "skeleton" Rails app.
- The following gems and their configuration:
  - Rspec
  - Factorybot
  - Faker
  - Database Cleaner
  - Shoulda Matchers
- Readme with instructions for 'Getting Started'

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+gem 'bootsnap', '>= 1.4.4', require: false
+
+# Use Rack CORS for handling Cross-Origin Resource Sharing (CORS), making cross-origin AJAX possible
+# gem 'rack-cors'
+
+group :development, :test do
+  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
+  gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
+  gem "factory_bot_rails"
+  gem "rspec-rails"
```
Comment (JorgeGarciaxyz) [inline Gemfile:32]: These two shouldn't be on `test` group?

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+end
+
+group :development do
+  gem "database_cleaner" 
+  gem "faker"
```
Comment (JorgeGarciaxyz) [inline Gemfile:37]: these and `shoulda_matchers` should be on the `test` group

File: README.md [hunk 3]
```diff
@@ -1 +1,52 @@
-# gifts_api_test
\ No newline at end of file
+# gifts_api_test
+
+This is a skeleton of a rails app to serve as a starting point for the code challence you are about to solve.
```
Comment (JorgeGarciaxyz) [inline README.md:3]:
```suggestion
This is a skeleton of a rails app to serve as a starting point for the code challenge you are about to solve.
```

File: config/application.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+require "active_record/railtie"
+require "active_storage/engine"
+require "action_controller/railtie"
+require "action_mailer/railtie"
+require "action_mailbox/engine"
+require "action_text/engine"
+require "action_view/railtie"
+require "action_cable/engine"
+# require "sprockets/railtie"
+require "rails/test_unit/railtie"
```
Comment (JorgeGarciaxyz) [inline config/application.rb:16]: I think this should be commented and delete the `test` folder, those are the minitest tests

File: config/credentials.yml.enc [hunk 5]
```diff
@@ -0,0 +1 @@
+GBgaXcvnJ/PVACI4TAmJoNiBUJ1sqh0eAoMNR8mh//mpyFdo4FCerrClcRszTHdKvEenmQXNmUdekmoWicKQEuJZ4zitpzPV84E17EmphdpQEQjHAVgyzE9v6AHXRhg1da2Ka+Jq/kSQE3bERQ1nTVy3Lx/KK0vHsstvmmAJ1d0xnr1gMtGl/2wPMJgxbTWZX+DcLrADUe/gMw8tqiTM0Sl7QLdIgWrUiMTUdWrdM0zwC9jaimfywz12GdYjbSaqFdBQWgXHLKh57VujOsarR103QhcUC3pLI7F8FyQRuAR7ojEdQGNGdRSc+pb2+TrDelF7tHkMXocQixx8SzQWbnvcCG0xxBny8zgQoEOlKhpPXFmi0C6MsFuJ18EJ8f5sf+9DUCvohfZ5Ctnh6Xpd3hssKlYZbV7rhZtc--qlS7qnQ+4S5H4mQk--yebmijknDUuDizQGnaniyg==
```
Comment (JorgeGarciaxyz) [inline config/credentials.yml.enc:1]: can you include the master key on the readme?

**Review by JorgeGarciaxyz — CHANGES_REQUESTED** [review]

File: config/credentials.yml.enc (hunk 5 above)
Comment (aecordova) [reply config/credentials.yml.enc:1]: removed the master key from the .gitignore

File: config/application.rb (hunk 4 above)
Comment (aecordova) [reply config/application.rb:16]: Done!

File: README.md (hunk 3 above)
Comment (aecordova) [reply README.md:3]: thanks, I've corrected this.

File: Gemfile (hunk 2 above)
Comment (aecordova) [reply Gemfile:37]: right, changed!

**Review by JorgeGarciaxyz — APPROVED** [review]

**Review by benoror — APPROVED** [review]

**Pull request merged**